### PR TITLE
feat: use lassie for retrievals

### PR DESCRIPTION
**Pull request by rvagg**
Not intended to be merged (yet anyway, we can have that conversation later). This is for dev purposes to make sure we get interfaces right and give me something functional so I can actually use code over in lassie for something real.

Depends on:

* https://github.com/filecoin-project/lassie/pull/4
* https://github.com/application-research/filclient/pull/112
  * https://github.com/filecoin-project/boost/pull/1009

### CODE REVIEW

**Comment by hannahhoward** [comment]
@rvagg @kylehuntsman 's PR is merged can we move this forward? Any reason not to?

**Comment by rvagg** [comment]
https://github.com/application-research/filclient/pull/112 is the blocker, and it's blocked by having a tagged boost—we could just use an untagged master over there but I don't think we need to rush this, I'm happy to keep this branch updated on both ends for now

**Review by elijaharita — APPROVED** [review]
looks great! dropping-in the current filclient on the retriever interface is a very clean transition to get started. if it's working, you have my blessing!

**Pull request merged**
